server/src/config.rs

```diff
@@ -0,0 +1,33 @@
+use std::env;
+
+use config::{ConfigError, Environment, File};
+use serde::Deserialize;
+
+#[derive(Clone, Deserialize)]
+pub struct Config {
+    pub session_secret: String,
+}
+
+impl Config {
+    /// Constructs a new [`Config`] by reading from multiple sources.
+    ///
+    /// In order of reading, the sources are:
+    ///
+    /// - `config/default.*`
+    /// - `config/$RUST_ENV.*`, defaults to `config/production.*` if `RUST_ENV` is undefined
+    /// - `config/local.*`
+    /// - Environment variables
+    ///
+    /// If a configuration value is defined in multiple sources, the value in a later source will
+    /// override the value in an earlier one.
+    pub fn new() -> Result<Self, ConfigError> {
+        let rust_env = env::var("RUST_ENV").unwrap_or_else(|_| "production".to_string());
+        config::Config::builder()
+            .add_source(File::with_name("config/default"))
```

Comment:
It looks like [`Config::set_default`](https://docs.rs/config/latest/config/builder/struct.ConfigBuilder.html#method.set_default) is meant for setting default configuration values individually rather than setting a default source, so I think it's probably best here to keep the defaults in a file. That's also what the crate seems to "recommend" in [their provided example](https://github.com/mehcode/config-rs/blob/master/examples/hierarchical-env/settings.rs#L50-L51).